src/measure_theory/decomposition/jordan.lean

```diff
@@ -0,0 +1,400 @@
+  neg_part := s.to_measure_of_le_zero iᶜ hi.1.compl hi.2.2,
+  pos_part_finite := infer_instance,
+  neg_part_finite := infer_instance,
+  mutually_singular :=
+  begin
+    refine ⟨iᶜ, hi.1.compl, _, _⟩,
+    { rw [to_measure_of_zero_le_apply _ _ hi.1 hi.1.compl], simpa },
+    { rw [to_measure_of_le_zero_apply _ _ hi.1.compl hi.1.compl.compl], simpa }
+  end }
+
+lemma to_jordan_decomposition_spec (s : signed_measure α) :
+  ∃ (i : set α) (hi₁ : measurable_set i) (hi₂ : 0 ≤[i] s) (hi₃ : s ≤[iᶜ] 0),
+    s.to_jordan_decomposition.pos_part = s.to_measure_of_zero_le i hi₁ hi₂ ∧
+    s.to_jordan_decomposition.neg_part = s.to_measure_of_le_zero iᶜ hi₁.compl hi₃ :=
+begin
+  set i := some s.exists_compl_positive_negative,
+  obtain ⟨hi₁, hi₂, hi₃⟩ := some_spec s.exists_compl_positive_negative,
+  exact ⟨i, hi₁, hi₂, hi₃, rfl, rfl⟩,
+end
+
+/-- **The Jordan decomposition theorem**: Given a signed measure `s`, there exists a pair of
+mutually singular measures `μ` and `ν` such that `s = μ - ν`. In this case, the measures `μ`
+and `ν` are given by `s.to_jordan_decomposition.μ` and `s.to_jordan_decomposition.ν` respectively.
+
+Note that we use `measure_theory.jordan_decomposition.to_signed_measure` to represent the
+signed measure corresponding to `s.to_jordan_decomposition.μ - s.to_jordan_decomposition.ν`. -/
+@[simp] lemma to_signed_measure_to_jordan_decomposition (s : signed_measure α) :
+  s.to_jordan_decomposition.to_signed_measure = s :=
+begin
+  obtain ⟨i, hi₁, hi₂, hi₃, hμ, hν⟩ := s.to_jordan_decomposition_spec,
+  simp only [jordan_decomposition.to_signed_measure, hμ, hν],
+  ext k hk,
+  rw [to_signed_measure_sub_apply hk, to_measure_of_zero_le_apply _ hi₂ hi₁ hk,
+      to_measure_of_le_zero_apply _ hi₃ hi₁.compl hk],
+  simp only [ennreal.coe_to_real, subtype.coe_mk, ennreal.some_eq_coe, sub_neg_eq_add],
+  rw [← of_union _ (measurable_set.inter hi₁ hk) (measurable_set.inter hi₁.compl hk),
+      set.inter_comm i, set.inter_comm iᶜ, set.inter_union_compl _ _],
+  { apply_instance },
+  { rintro x ⟨⟨hx₁, _⟩, hx₂, _⟩,
+    exact false.elim (hx₂ hx₁) }
+end
+
+section
+
+variables {u v w : set α}
+
+/-- A subset `v` of a null-set `w` has zero measure if `w` is a subset of a positive set `u`. -/
+lemma subset_positive_null_set
+  (hu : measurable_set u) (hv : measurable_set v) (hw : measurable_set w)
+  (hsu : 0 ≤[u] s) (hw₁ : s w = 0) (hw₂ : w ⊆ u) (hwt : v ⊆ w) : s v = 0 :=
+begin
+  have : s v + s (w \ v) = 0,
+  { rw [← hw₁, ← of_union set.disjoint_diff hv (hw.diff hv),
+        set.union_diff_self, set.union_eq_self_of_subset_left hwt],
+    apply_instance },
+  have h₁ := nonneg_of_zero_le_restrict _ (restrict_le_restrict_subset _ _ hu hsu (hwt.trans hw₂)),
+  have h₂ := nonneg_of_zero_le_restrict _
+    (restrict_le_restrict_subset _ _ hu hsu ((w.diff_subset v).trans hw₂)),
+  linarith,
+end
+
+/-- A subset `v` of a null-set `w` has zero measure if `w` is a subset of a negative set `u`. -/
+lemma subset_negative_null_set
```

Comment:
you have a lot of lemmas that you know in the positive case, and that you reprove in the negative case. Don't you get a shorter proof by applying the positive result to `-s`? (Here and several times below).